Sources/AppcuesKit/Data/Models/Experiment.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  Experiment.swift
+//  AppcuesKit
+//
+//  Created by James Ellis on 10/17/22.
+//  Copyright © 2022 Appcues. All rights reserved.
+//
+
+import Foundation
+
+internal struct Experiment: Decodable {
+    let group: String
+}
+
+extension Experiment {
+    func shouldExecute() -> Bool {
+        return group != "control"
+    }
+
+    func track(appcues: Appcues?, experimentID: String?) {
+        guard let experimentID = experimentID,
+              let analyticsPublisher = appcues?.container.resolve(AnalyticsPublishing.self) else { return }
+
+        analyticsPublisher.publish(TrackingUpdate(
+            type: .event(name: "appcues:experiment_entered", interactive: false),
+            properties: [
+                "experimentId": experimentID,
+                "group": group
+            ],
+            isInternal: true))
+    }
```

Comment:
these are now used during the `ExperienceRenderer.show(...)` processing, encapsulating usage of `AnalyticsPublishing` here 🤷‍♂️

this has to be two-phase - `shouldExecute` then `track` - explained in the comments in usage spot in `ExperienceRenderer.show(...)`

Reply:
It does feel a bit off having the tracking in a model struct just to avoid the dependency graph change 🤷‍♂️

There's already a dependency from `AnalyticsObserver` to `AnalyticsTracking` so I could just move that line over to start from `ExperienceRendering`. I think doing this and then tracking from the `ExperienceRender` might be preferable for me than having the tracking in the model, but final call is yours.

Reply:
I agree, updated - thanks.

Reply:
Updated the graph:
![iOS SDK Dependency Graph](https://user-images.githubusercontent.com/845681/197247167-eb29c7da-9495-4483-8418-0e6a856200d8.png)

Reply:
thank you!